opensoundscape/metrics.py

```diff
@@ -81,18 +81,20 @@ def compute_epoch_metrics(self):
         recalls = [None] * self.num_classes
         f1s = [None] * self.num_classes
 
-        num_observations = self.confusion_matrix.sum()
         for idx, cls in enumerate(self.classes):
-            row_observations = self.confusion_matrix[idx, :].sum()
-            col_observations = self.confusion_matrix[:, idx].sum()
+            class_observations = self.confusion_matrix[idx, :].sum()
+            class_predictions = self.confusion_matrix[:, idx].sum()
 
             true_positives = self.confusion_matrix[idx, idx]
             # row + col observations double subtract true_positives
             true_negatives = (
-                num_observations - row_observations - col_observations + true_positives
+                self.confusion_matrix.sum()
```

Comment:
Why sum this every time in the loop? Could restore the definition of this variable outside the loop, but call it something different if it is a variable name issue, e.g. `total_observations = self.confusion_matrix.sum()`

Reply:
Ahh, this was a mistake. I thought, "If I inline this than I don't have to name it poorly". However, this would get recomputed every time which is an issue.